test/runtime/fqdn.go

```diff
@@ -331,6 +332,64 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
 		res.ExpectSuccess("Cannot access to allowed DNS name %q", world1Target)
 	})
 
+	It("Validate dns-proxy monitor information", func() {
+		policy := `
+[
+	{
+		"labels": [{
+			"key": "monitor"
+		}],
+		"endpointSelector": {
+			"matchLabels": {
+				"container:id.app1": ""
+			}
+		},
+		"egress": [
+			{
+				"toPorts": [{
+					"ports":[{"port": "53", "protocol": "ANY"}],
+					"rules": {
+						"dns": [
+							{"matchPattern": "world1.cilium.test"}
+						]
+					}
+				}]
+			},
+			{
+				"toFQDNs": [{
+					"matchPattern": "world1.cilium.test"
+				}]
+			}
+		]
+	}
+]`
+		_, err := vm.PolicyRenderAndImport(policy)
+		Expect(err).To(BeNil(), "Policy cannot be imported")
+
+		expectFQDNSareApplied()
+
+		ctx, cancel := context.WithCancel(context.Background())
+		monitorCMD := vm.ExecInBackground(ctx, "cilium monitor")
+		defer cancel()
+
+		allowVerdict := "verdict Forwarded DNS Query: world1.cilium.test"
+		deniedVerdict := "verdict Denied DNS Query: world2.cilium.test"
+
+		By("Testing connectivity to Cilium.test domain")
+		res := vm.ContainerExec(helpers.App1, helpers.CurlFail(world1Target))
+		res.ExpectSuccess("Cannot access to %q", world1Target)
+
+		_ = monitorCMD.WaitUntilMatch(allowVerdict)
```

Comment:
We should wrap all these with a check that the returned error is nil. Otherwise, when the monitor output is incorrect, we would timeout the whole CI instead of failing this test, right?

Reply:
I would still say an explicit timeout error is more clear than "allow was missing in this string, which happens to be empty". I think it will be ok anyway, though :)